### ISSUE TITLE
Template email links result in error when clicked in archive

### ISSUE THREAD

**Issue by cebner-proxmox** [knadh/listmonk#1506]
**Version:**
 - listmonk: v2.5.1

**Description of the bug and steps to reproduce:**
All links created from a campaign template using subscriber related template variables, e.g. `{{ UnsubscribeURL }}` or view in browser `{{ MessageURL }}` result in links producing a json error response in the browser, when the user tries to click the link viewing it from the newsletter archive.

It would be desirable, if these links either redirect to some valid content or are not clickable if viewed via the newsletter archive.

**Comment by knadh** [comment]
Can you share the actual URL that opens? Does it point to the correct domain?

**Comment by cebner-proxmox** [comment]
Thanks for the reply @knadh.
Please follow this URL to find an example, where clicking the link produces the error [https://newsletter.proxmox.com/campaign/1345df93-c713-45d0-a877-145b54d1f2e6/](https://newsletter.proxmox.com/campaign/1345df93-c713-45d0-a877-145b54d1f2e6/)
The corresponding Email in the archive can be found here [https://newsletter.proxmox.com/archive/1345df93-c713-45d0-a877-145b54d1f2e6](https://newsletter.proxmox.com/archive/1345df93-c713-45d0-a877-145b54d1f2e6), both, the view in browser and the unsubscribe link in the Email lead to errors in the archive, while working as expected in the Emails send to the subscribers.

Please let me know if I can provide further insights or configurations to help debug the issue, thx.

**Comment by etranger7** [comment]
Quick question about this @cebner-proxmox:  
Is this link working in the email body but not working in the public archive?  
Or is it broken in both?

Within the same campaign, it's possible to define a separate template for the emails as opposed to the public archive page so that's why I'm wondering which one causing the issue here and whether there is a discrepancy. It might also be useful if you posted the code of that template.

**Comment by cebner-proxmox** [comment]
Hi @etranger7,
The links are broken only in the archive, within the emails they work as expected. This was reproduced with the default template provided by listmonk (I did not set a different one for the archive).
Indeed I overlooked the option to set the different template for the campaign archive. So is the suggested workflow here to duplicate the template information, one for email and one for the archive and remove this links in the archive template?

If that is the case, than this is indeed just a misunderstand/oversight of how the template system is intended to work from my side.
